src/main/java/org/reactome/addlinks/dataretrieval/ensembl/EnsemblBiomartRetriever.java

```diff
@@ -0,0 +1,232 @@
+
+    private String getMicroarrayTypesQuery(String biomartSpeciesName) {
+        return this.getDataURL() + microarrayTypesBaseQuery.replace("BIOMART_SPECIES_NAME", biomartSpeciesName);
+    }
+
+    /**
+     * Query Biomart for Microarray 'identifiers' associated with the microarray type
+     * @param biomartSpeciesName -- String of species name in Biomart format (eg: hsapiens)
+     * @param microarrayTypes -- String, type of microarray identifiers being queried for.
+     * @throws Exception
+     */
+    private void retrieveAndStoreMicroarrayData(String biomartSpeciesName, String biomartXMLPath, Set<String> microarrayTypes) {
+
+        // Query Biomart for microarray identifiers associated with microarray type.
+        for (String microarrayType : microarrayTypes) {
+            try {
+                logger.info("Retrieving data associated with microarray type " + microarrayType);
+                Set<String> microarrayResponseLines = queryBiomart(getBiomartIdentifierQuery(biomartXMLPath, biomartSpeciesName, microarrayType), 0);
+                Set<String> microarrayIdentifierLines = new HashSet<>();
+                for (String microarrayResponseLine : microarrayResponseLines) {
+                    List<String> tabSplit = Arrays.asList(microarrayResponseLine.split("\t"));
+                    // Microarray ID is 4th column, and sometimes it isn't in a line.
+                    // Since we need the ID, we require the List to have 4 elements.
+                    if (tabSplit.size() == 4) {
+                        microarrayIdentifierLines.add(microarrayResponseLine + "\n");
+                    }
+                }
+
+                // Write data to file.
+                if (!microarrayIdentifierLines.isEmpty()) {
+                    File speciesMicroarrayFile = createNewFile(this.destination + biomartSpeciesName + "_microarray");
+                    for (String probeIdLine : microarrayIdentifierLines) {
+                        Files.write(Paths.get(speciesMicroarrayFile.toString()), probeIdLine.getBytes(), StandardOpenOption.APPEND);
+                    }
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Attempts to create a connection to Biomart with the supplied query
+     * @param queryString String of Biomart query
+     * @return Set<String> Biomart response data.
+     * @throws Exception Can be caused by the query URL not being valid, or the data not existing for the species.
+     */
+    private Set<String> queryBiomart(String queryString, int retryCount) throws Exception {
+        // Create connection to Biomart URL for each species, retrieving a list of microarray probe types, if available.
+        URL biomartUrlWithSpecies = new URL(queryString);
+        HttpURLConnection biomartConnection = (HttpURLConnection) biomartUrlWithSpecies.openConnection();
+        if (biomartConnection.getResponseCode() == 200) {
```

Comment:
If you want to use response code _names_, I think there's a class `HttpStatus`. So you can do 
```java
biomartConnection.getResponseCode() == HttpStatus.SC_OK
```
...it's more useful when you're checking for more obscure response codes that are not as well know.

(https://hc.apache.org/httpclient-3.x/apidocs/org/apache/commons/httpclient/HttpStatus.html#SC_OK)